src/main/java/bio/terra/service/dataset/AzureStorageResource.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.service.dataset;
+
+import bio.terra.app.model.AzureCloudResource;
+import bio.terra.app.model.AzureRegion;
+import bio.terra.app.model.CloudRegion;
+import bio.terra.app.model.CloudResource;
+import bio.terra.model.CloudPlatform;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import java.util.Objects;
+import java.util.UUID;
+
+@JsonTypeName("azure")
+public class AzureStorageResource implements StorageResource {
+
+    private UUID datasetId;
+    private CloudPlatform cloudPlatform;
+    private AzureCloudResource cloudResource;
+    private AzureRegion region;
+
+    AzureStorageResource() {
+        this.cloudPlatform = CloudPlatform.AZURE;
```

Comment:
This could be done using a field initializer, and the field can be `final`. Or, you can remove the field and change `getCloudPlatform()` to `return CloudPlatform.AZURE;`

Reply:
Yeah, that's a fair point...and we don't need it in the equality check since it's bound to the class type